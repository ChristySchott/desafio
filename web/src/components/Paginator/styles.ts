import styled from 'styled-components';

export const Wrapper = styled.section`
  align-items: center;
  display: flex;
  padding: 1.5rem 3rem;
  justify-content: space-between;
  max-width: 350px;
  margin: 12px auto;

  a {
    text-decoration: none;
    transition: color 0.5s;
    height: 20px;
  }
`;

export const Switch = styled.ul`
  display: flex;
  align-items: center;
  justify-content: space-around;
  flex: 3;
  height: 100%;
`;

export const Page = styled.a`
  font-size: 1.5rem;
  line-height: 16px;
  font-weight: 600;
  letter-spacing: 0.914285px;
  cursor: pointer;

  &:hover {
    color: #c61a28;
  }
`;

export const PaginatorLink = styled.a``;
