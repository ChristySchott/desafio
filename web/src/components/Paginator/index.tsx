import React from 'react';

import { FiChevronLeft, FiChevronRight } from 'react-icons/fi';

import { Wrapper, Switch, Page, PaginatorLink } from './styles';

const Paginator: React.FC = () => {
  return (
    <>
      <Wrapper>
        <PaginatorLink>
          <FiChevronLeft size={20} color="#5dbcd2" />
        </PaginatorLink>
        <Switch>
          <Page>1</Page>
        </Switch>
        <PaginatorLink>
          <FiChevronRight size={20} color="#5dbcd2" />
        </PaginatorLink>
      </Wrapper>
    </>
  );
};

export default Paginator;
