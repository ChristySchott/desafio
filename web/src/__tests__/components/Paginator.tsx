/* eslint-disable no-console */
import React from 'react';
import { render } from '@testing-library/react';

import Paginator from '../../components/Paginator';

describe('<Paginator />', () => {
  it('should render with no error', () => {
    const { container } = render(<Paginator />);

    expect(container).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
