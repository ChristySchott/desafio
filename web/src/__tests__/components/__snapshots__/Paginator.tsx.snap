// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paginator /> should render with no error 1`] = `
<section
  class="sc-bdnylx cFSjuR"
>
  <a
    class="sc-hKFyIo"
  >
    <svg
      color="#5dbcd2"
      fill="none"
      height="20"
      size="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      style="color: rgb(93, 188, 210);"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="15 18 9 12 15 6"
      />
    </svg>
  </a>
  <ul
    class="sc-gtssRu kscDjb"
  >
    <a
      class="sc-dlnjPT dJGKuS"
    >
      1
    </a>
  </ul>
  <a
    class="sc-hKFyIo"
  >
    <svg
      color="#5dbcd2"
      fill="none"
      height="20"
      size="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      style="color: rgb(93, 188, 210);"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="9 18 15 12 9 6"
      />
    </svg>
  </a>
</section>
`;
